pkg/signature/kms/azure/client.go

```diff
@@ -304,21 +304,47 @@ func (a *azureVaultClient) createKey(ctx context.Context) (crypto.PublicKey, err
 	return a.public(ctx)
 }
 
-func getKeyVaultSignatureAlgo(algo crypto.Hash) (keyvault.JSONWebKeySignatureAlgorithm, error) {
-	switch algo {
+func (a *azureVaultClient) getKeyVaultHashFunc() (crypto.Hash, error) {
```

Comment:
Maybe outside of scope for this PR, but should this client support more than just ECDSA keys? Let's at least document this somewhere (fwiw every client is a little different in this regard)

Reply:
GCP has the same for KMS keys, it looks like there's a hardcoded mapping between key name and hash algorithm - https://github.com/sigstore/sigstore/blob/main/pkg/signature/kms/gcp/client.go#L210-L232

Do we want to do the same here? Another benefit is it makes it clear which key types we support.

Reply:
So I started digging into this more. `SignMessage` is just a sigstore/sigstore interface, and not one called from `crypto/x509` `x509.CreateCertificate`, which calls `Sign` with a given digest. It looks like for RSA, it's hardcoded for sha256 - https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/crypto/x509/x509.go;l=1410

I was surprised to see this. I might be wrong, I haven't dug into this much, but if so, then it's relevant if you provide a different hash function for RSA.

Reply:
Sorry, completely ignore what I wrote above, that's just the default.

But we do still need to make a change in Fulcio to specify the SignatureAlgorithm.

Reply:
https://github.com/sigstore/fulcio/issues/1161 

We should probably expose the signature algorithm from sigstore/sigstore too.